tensorboard/plugins/text/text_plugin.py

```diff
@@ -231,106 +212,25 @@ def is_active(self):
-      return True
-
-    # We haven't conclusively determined if the plugin should be active. Launch
-    # a thread to compute index_impl() and return False to avoid blocking.
-    self._maybe_launch_index_impl_thread()
-
-    return False
+    return bool(self._multiplexer.PluginRunToTagToContent(metadata.PLUGIN_NAME))
 
   def frontend_metadata(self):
     return super(TextPlugin, self).frontend_metadata()._replace(
         element_name='tf-text-dashboard',
     )
 
-  def _maybe_launch_index_impl_thread(self):
-    """Attempts to launch a thread to compute index_impl().
-
-    This may not launch a new thread if one is already running to compute
-    index_impl(); in that case, this function is a no-op.
-    """
-    # Try to acquire the lock for computing index_impl(), without blocking.
-    if self._index_impl_lock.acquire(False):
-      # We got the lock. Start the thread, which will unlock the lock when done.
-      self._index_impl_thread = threading.Thread(
-          target=self._async_index_impl,
-          name='TextPluginIndexImplThread')
-      self._index_impl_thread.start()
-
-  def _async_index_impl(self):
-    """Computes index_impl() asynchronously on a separate thread."""
-    start = time.time()
-    logger.info('TextPlugin computing index_impl() in a new thread')
-    self._index_cached = self.index_impl()
-    self._index_impl_thread = None
-    self._index_impl_lock.release()
-    elapsed = time.time() - start
-    logger.info(
-        'TextPlugin index_impl() thread ending after %0.3f sec', elapsed)
-
   def index_impl(self):
-    run_to_series = self._fetch_run_to_series_from_multiplexer()
-
-    # A previous system of collecting and serving text summaries involved
-    # storing the tags of text summaries within tensors.json files. See if we
-    # are currently using that system. We do not want to drop support for that
-    # use case.
-    name = 'tensorboard_text'
-    run_to_assets = self._multiplexer.PluginAssets(name)
-    for run, assets in run_to_assets.items():
-      if run in run_to_series:
-        # When runs conflict, the summaries created via the new method override.
-        continue
-
-      if 'tensors.json' in assets:
-        tensors_json = self._multiplexer.RetrievePluginAsset(
-            run, name, 'tensors.json')
-        tensors = json.loads(tensors_json)
-        run_to_series[run] = tensors
-      else:
-        # The mapping should contain all runs among its keys.
-        run_to_series[run] = []
-
-    return run_to_series
-
-  def _fetch_run_to_series_from_multiplexer(self):
-    # TensorBoard is obtaining summaries related to the text plugin based on
-    # SummaryMetadata stored within Value protos.
-    mapping = self._multiplexer.PluginRunToTagToContent(
-        metadata.PLUGIN_NAME)
+    mapping = self._multiplexer.PluginRunToTagToContent(metadata.PLUGIN_NAME)
     return {
-        run: list(tag_to_content.keys())
+        run: list(six.iterkeys(tag_to_content))
```

Comment:
No reason to use `iterkeys()` only to pass it to `list()` right?  Just `list(tag_to_content)`

Reply:
Done; thanks.